flutter/ARCHITECTURE.md

```diff
@@ -484,32 +461,15 @@ signature is: `showExceptionSnackBar(BaseException exception)`.
 
 ## App Icon
 
-The application icon is auto-generated using a high-res single icon asset file alongside a commonly used package that
-uses this asset to generate for all platforms (`flutter-launcher-icons`).
-
-If you change the icon's asset, you have to run this tool again to make sure that the files are updated:
-
-```
-flutter pub run flutter_launcher_icons:main
-```
-
-## Local collections and resources versioning
-
-Because the current `memo` version only manages its data locally, it has no server to defer more complex use-cases, like
-syncing the user's list of `Collection`s (with its `Memo`s) and its `Resource`s.
+The icons are auto-generated using a high-res asset file by package (`flutter-launcher-icons`) that does this for all platforms.
 
-This sync is done whenever a new application version is deployed - by checking which was the last stored version. Then,
-when this version differs from the last stored one, runs through all the `assets/collections/` files to make the
-according additions, updates and removals. The same happens to `assets/resources.json`, but in this case, there is a
-single json file containing all the resources available to the application.
-
-This version check runs once every time the application opens, but the heavy computation only occurs when the current
-versus expected version differs. You can find more about this in the `VersionServices` implementation file.
+If you change the asset, you'll have to run this tool again to update the required files:
+`flutter pub run flutter_launcher_icons:main`
 
 ## Environment
 
-For different types of build environments, we don't use the common _flavors_, iOS schemas and all of that painful setup,
-due to the fact that, since Flutter `1.17`, we can now use command arguments to inject any variable in our application -
+For different types of build environments, we **don't** use flavors, iOS schemas and all of that painful setup, due to
+the fact that, since Flutter `1.17`, we can now use command arguments to inject any variable in our application -
```

Comment:
This is going to change when adding Firebase, isn't it? If using different containers to split development and production environments in Firebase, we'll need to use flutter flavors because we'll have different iOS bundle identifier and Android package identifiers for development and production respectively

Reply:
I don't think so, we won't have development/production environments. We will possible have only local and production, and this could possible still coexist with using only CLI args

Reply:
But we can make some definitions today. Using Firebase Emulators for development without a development container in Firebase console has known impacts today, and it's necessary to decide if we are going to face them or not. I'll list some of them below

- [Firebase Emulator limitations](https://firebase.google.com/docs/emulator-suite/connect_firestore#limitations)
- Storage emulator is barely unuseful because it was recently launched and have tons of [bugs and errors](https://github.com/firebase/firebase-tools/issues?q=is%3Aopen+is%3Aissue+label%3A%22emulators%3A+storage%22)

Just to be clear: I'm not saying about the docs, I'm making this statements because their answer may change the server-side development in the short-term. I'm ok with updating the docs later (or not if we are going to use Firebase Emulators only), but we have to define if we need a development container or not.

Reply:
> Firebase Emulator limitations

I'm fine with these today. If we do need these indexes, we can think about adding a whole environment setup in the future.

> Storage emulator is barely useful* because it was recently launched and have tons of bugs and errors

Not used today, and, if necessary, the same logic above applies to this.